src/MetaModels/Attribute/Checkbox/Checkbox.php

```diff
@@ -93,35 +93,19 @@ public function getFieldDefinition($arrOverrides = array())
      */
     public function getFilterOptions($idList, $usedOnly, &$arrCount = null)
     {
-        $values = parent::getFilterOptions($idList, $usedOnly, $arrCount);
-        // If not used only or id list, we need to override the handling now.
-        if (!($idList || $usedOnly)) {
-            $values = ['', '1'];
-            if (null !== $arrCount) {
-                $arrCount[''] = 0;
-                $arrCount[1]  = 0;
-                $colName      = $this->getColName();
-                $rows         = $this->getDatabase()->execute(
-                    sprintf(
-                        'SELECT %1$s, COUNT(%1$s) as mm_count FROM %2$s GROUP BY %1$s ORDER BY %1$s',
-                        $colName,
-                        $this->getMetaModel()->getTableName()
-                    )
-                );
-                while ($rows->next()) {
-                    $arrCount[$rows->$colName] = $rows->mm_count;
-                }
-            }
+        
+        $arrOptions = parent::getFilterOptions($idList, $usedOnly, $arrCount);
+        foreach ($arrOptions as $key => $value) {
+           if ($key == '1') {
+              $arrOptions['1'] = $GLOBALS['TL_LANG']['MSC']['metamodelattribute_checkbox']['value_1'];
+           }else{
+            unset($arrOptions[$key]);
+            $arrOptions['0'] = $GLOBALS['TL_LANG']['MSC']['metamodelattribute_checkbox']['value_0'];
+           }
         }
+           
+        return $arrOptions;
 
-        // Finally use the correct language strings.
```

Comment:
@discodier 
1. yes you are right. the '0' is a mistake. 
2. ok, what I don't really understand is why we have to count the amount of options. It's always a single checkbox with two options. And I suppose the Databasevalue of this attribute will only have the content : null, '', or '1'. So if a value is not a (string) 1 there went something wrong while the save action. Or how should the number 1 get into the database field?